### PR TITLE
Feature/acm observability

### PR DESCRIPTION
**Pull request by computate**
Sets up the read-only ACM Observability tools for analyzing the default dashboards in Grafana
image
image

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202099997018489

### CODE REVIEW

**Review by hpdempsey — COMMENTED** [review]
@dystewart can you please review the cert and Kustomize parts of this to make sure the base is done according to guidelines? I'm not familiar enough with that to review.

@computate There are several settings for intervals, so I wanted to review to make sure they were consistent for how often we were sampling, storing, expiring certs etc.  I searched a online but couldn't find the appropriate APIs that define the fields easily.  Can you point me to the API definitions for the variables that control various timing settings please?

I assume that observability downsampling is enabled by default, right?  We may not want to do this in the end, because we are also generating a dataset for researchers, so preserving the full dataset may be preferable.  It is a tradeoff between content and storage.  However, if this is the default, I think it is fine to leave it that way while we work through initial deployment and testing.

File: cluster-scope/base/external-secrets.io/externalsecrets/open-cluster-management-observability-multiclusterhub-operator-pull-secret/externalsecret.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+spec:
+  refreshInterval: 15s
+  secretStoreRef:
+    name: nerc-cluster-secrets
+    kind: ClusterSecretStore
+  target:
+    name: multiclusterhub-operator-pull-secret
+  dataFrom:
+    - extract:
+        key: $ENV/$CLUSTER/open-cluster-management-observability-multiclusterhub-operator-pull-secret
```
Comment (naved001) [inline cluster-scope/base/external-secrets.io/externalsecrets/open-cluster-management-observability-multiclusterhub-operator-pull-secret/externalsecret.yaml:16]: you need to patch this key `$ENV/$CLUSTER/...` in the overlay for nerc-ocp-prod

File: cluster-scope/base/external-secrets.io/externalsecrets/open-cluster-management-observability-thanos-object-storage/externalsecret.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    remoteRef:
+      key: $ENV/$CLUSTER/open-cluster-management-observability-thanos-object-storage
+      property: bucket
+  - secretKey: access_key
+    remoteRef:
+      key: $ENV/$CLUSTER/open-cluster-management-observability-thanos-object-storage
+      property: access_key
+  - secretKey: secret_key
+    remoteRef:
+      key: $ENV/$CLUSTER/open-cluster-management-observability-thanos-object-storage
```
Comment (naved001) [inline cluster-scope/base/external-secrets.io/externalsecrets/open-cluster-management-observability-thanos-object-storage/externalsecret.yaml:46]: the 3 references to key will also need to be patched in the overlay

Comment (hpdempsey) [reply cluster-scope/base/external-secrets.io/externalsecrets/open-cluster-management-observability-thanos-object-storage/externalsecret.yaml:46]: Thanks @naved001

**Comment by computate** [comment]
@hpdempsey I found some documentation on the timing configuration that might help: 

https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.5/html/observability/observing-environments-intro#adding-advanced-config

https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.5/html/apis/apis#rhacm-docs_apis_multiclusterobservability_jsonmulticlusterobservability

File: cluster-scope/base/external-secrets.io/externalsecrets/open-cluster-management-observability-multiclusterhub-operator-pull-secret/externalsecret.yaml (hunk 1 above)
Comment (computate) [reply cluster-scope/base/external-secrets.io/externalsecrets/open-cluster-management-observability-multiclusterhub-operator-pull-secret/externalsecret.yaml:16]: Thanks @naved001 I have patched these values for nerc-ocp-prod in the latest commits.

File: cluster-scope/base/external-secrets.io/externalsecrets/open-cluster-management-observability-thanos-object-storage/externalsecret.yaml (hunk 2 above)
Comment (computate) [reply cluster-scope/base/external-secrets.io/externalsecrets/open-cluster-management-observability-thanos-object-storage/externalsecret.yaml:46]: Thanks @naved001 @hpdempsey  I have patched these values for nerc-ocp-prod in the latest commits.

**Review by dystewart — COMMENTED** [review]
@computate I think everything looks good, is all the secret data in Vault already?

**Comment by computate** [comment]
@dystewart yes, the secret data is in vault.

**Review by dystewart — APPROVED** [review]

**Pull request merged**